### DISCUSSION TITLE
Error boundary or custom error page from middlewares

### DISCUSSION THREAD

**Discussion by stefanosandes** [sergiodxa/remix-hono#178] [Q&A]
Hello!

Is it possible to set up error pages using Hono middleware? I want to avoid redirects or URL changes. 
My use case is as follows: I have a middleware that checks for a condition. I want to proceed or show a "not found" message depending on the condition.

Thanks!

**Comment by sergiodxa** [comment]
Why not use Remix's ErrorBoundary component?

**Reply by stefanosandes** [reply]
I think the ErrorBoundary will only work if the error was thrown from the tree. Am I wrong? Errors thrown from a Hono middleware do not reach the ErrorBoundary.
